### PR TITLE
Feature Implementation of GCN results

### LINKED ISSUE #11: Results : GCN
Report training and inference results for analysis for the GCN, adjust hyperparameters as needed. Make sure to document any changes to hyperparameters.
- Make note of parameters and hyperparameters used.
- Report training accuracy and loss across epochs
- Report confusion matrix
- Report accuracy metrics
- Report visual of true label graph vs. predicted label graph

### PR DESCRIPTION
**Pull request by BrandonN24**
Closes #11 

Features:
- plot_acc.py
- plot_confusion_matrix.py
- refactored test_model() function in training/GCN_training.py to utils/test_model.py
- Implemented all metrics from issue #11 except "Report visual of true label graph vs. predicted label graph".

IMPORTANT: Merge #17 into main before merging this PR!

### CODE REVIEW

**Review by simonbruce85 — APPROVED** [review]
lgtm
